docs/content/docs/connectors/datastream/kinesis.md

```diff
@@ -566,124 +583,124 @@ Retry and backoff parameters can be configured using the `ConsumerConfigConstant
+The sink default maximum record size is 1MB and maximum batch size is 5MB in line with the Kinesis Data Streams maximums.
 
-Users can still switch back to one-thread-per-request mode by setting a key-value pair of `ThreadingModel` and `PER_REQUEST` in `java.util.Properties`, as shown in the code commented out in above example.
+### Kinesis Sinks and Fault Tolerance
 
-### Backpressure
+The sink is designed to participate in Flink's checkpointing to provide at-least-once processing guarantees. It does this by flushing the entire contents of the buffer when a checkpoint reaches the sink. This effectively assures all requests that were triggered before the checkpoint have been successfully acknowledged by Kinesis Data Streams, before proceeding to process more records sent to the sink.
 
-By default, `FlinkKinesisProducer` does not backpressure. Instead, records that
-cannot be sent because of the rate restriction of 1 MB per second per shard are
-buffered in an unbounded queue and dropped when their `RecordTtl` expires.
+In case of a failure or a resharding, data will be written again to Kinesis, leading to duplicates.  Also, the sink does not guarantee that records are written in order to the shards.
 
-To avoid data loss, you can enable backpressuring by restricting the size of the
-internal queue:
+To use fault tolerant Kinesis Sinks, checkpointing of the topology needs to be enabled at the execution environment.
 
-```
-// 200 Bytes per record, 1 shard
-kinesis.setQueueLimit(500);
-```
+### Backpressure
 
-The value for `queueLimit` depends on the expected record size. To choose a good
-value, consider that Kinesis is rate-limited to 1MB per second per shard. If
-less than one second's worth of records is buffered, then the queue may not be
-able to operate at full capacity. With the default `RecordMaxBufferedTime` of
-100ms, a queue size of 100kB per shard should be sufficient. The `queueLimit`
-can then be computed via
+Backpressure in the sink arises as the sink buffer fills up and writes to the sink 
+begins to exhibit blocking behaviour. Kinesis Data Streams has a rate restriction of 
+1 MB per second per shard.
 
+You can ease backpressure by adjusting the size of the internal queue:
 ```
-queue limit = (number of shards * queue size per shard) / record size
+KinesisDataStreamsSink<String> kdsSink =
+    KinesisDataStreamsSink.<String>builder()
+        ...
+        .setMaxBufferedRequests(10_000)
```

Comment:
We can also use `setMaxBatchSizeInBytes` ?

nit: Again, this will be common for all async sink and we should consider moving it somewhere general

Reply:
I'm not sure there is such a location. The community previously agreed in an earlier PR that we should not have a documentation page for the Async Sink. Another candidate might be the pages on backpressure in the documentation(such as https://nightlies.apache.org/flink/flink-docs-master/docs/ops/monitoring/back_pressure/), but I imagine I would have to be explicit about the fact that this only affects the sinks based on the async sink...

I'm not sure WDYT?

(I'll add `setMaxBatchSizeInBytes`)

Reply:
Moving the docs to a common place is lower priority for now, we can refactor with the Firehose sink, let's focus on documentation quality for this PR. 

What does `ease backpressure` mean? Do you mean if you are backpressured due to a large number of small sized messaged you can increase the throughput to reduce backpressure? Is it worth noting that the Kinesis limits are per shard and depending on the shard:parallelism ratio you may have a much bigger limit? For example, 1 subtask writing to 10 shards can achieve 10MBps for uniformly distributed data.

Reply:
I just wanted `ease backpressure` here to mean that if you increased the `maxBufferedRequests`, the buffer would be less frequently full and so when the sink operator calls `write`, the sink will block less frequently.

Reply:
This indicates the default configuration may unnecessarily apply backpressure? This is concerning...

Reply:
As discussed offline, this is not an issue with docs. We can change this if/when we change behaviour